### ISSUE TITLE
Incompatibility with from __future__ import annotations

### ISSUE THREAD

**Issue by djbower** [ajatkj/typed_configparser#9]
### Description

If you add `from __future__ import annotations` to the top of the example code snippets the code fails.

### Example Code

```Python
# This is a complete example and should work as is

# Extra line here breaks typed_configparser
from __future__ import annotations

from typing import List
from typed_configparser import ConfigParser
from dataclasses import dataclass


@dataclass
class BASIC:
    option1: int
    option2: str
    option3: float
    option4: List[str]


config = """
[BASIC]
option1 = 10
option2 = value2
option3 = 5.2
option4 = [foo,bar]
"""

parser = ConfigParser()
parser.read_string(config)
section = parser.parse_section(using_dataclass=BASIC)

print(section)
```

### Python, typed_configparser & OS Version

```Text
Python 3.10.13, OSX, typed parser 1.0.2
```

**Comment by ajatkj** [comment]
Thanks for reporting this. I think I get it. I was using annotation dictionary instead of get_type_hints to get the type hints. Should be fixed in next release.

**Comment by ajatkj** [comment]
Changes released in version 1.1.0 (also available on pypi)
